keycloak-extension-rest/src/main/java/de/conciso/keycloak/resource/admin/GetUsersByIdResource.java

```diff
@@ -0,0 +1,56 @@
+package de.conciso.keycloak.resource.admin;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.Status;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.jboss.logging.Logger;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.utils.ModelToRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.services.resources.admin.permissions.AdminPermissionEvaluator;
+
+public class GetUsersByIdResource {
+
+  private static final Logger log = Logger.getLogger(GetUsersByIdResource.class);
+  private final KeycloakSession session;
+  private final AdminPermissionEvaluator auth;
+
+  public GetUsersByIdResource(KeycloakSession session, AdminPermissionEvaluator auth) {
+    this.session = session;
+    this.auth = auth;
+  }
+
+
+  @GET
+  @Produces({MediaType.APPLICATION_JSON})
+  public Response getAllUsersByListOfIds(
+      @QueryParam("listWithIds") List<UUID> list,
+      @QueryParam("briefRepresentation") boolean briefRepresentation) {
+    auth.users().requireQuery();
+
+    List<UserRepresentation> userRepresentations = new ArrayList<>();
+    for (UUID id : list) {
+      final var userModel = session
```

Comment:
This lookup won't work for lightweight/transient users which are a new feature since KC 23.x.

Check the `UsersResource` code at https://github.com/keycloak/keycloak/blob/24.0.3/services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java#L213 for details and how to support transient users.

Reply:
updated, thanks for the hint !